### PR TITLE
Deprecate in_place in clear_border

### PR DESCRIPTION
**Pull request by rfezzani**
## Description

This PR deprecates `in_place` argument in `clear_border`

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: skimage/segmentation/_clear_border.py [hunk 1]
```diff
@@ -58,49 +67,54 @@ def clear_border(labels, buffer_size=0, bgval=0, in_place=False, mask=None):
...
+            borders[tuple(slices)] = True
+            slices[d] = slend
+            borders[tuple(slices)] = True
+            slices[d] = slice(None)
+
+    # Re-label, in case we are dealing with a binary out
     # and to get consistent labeling
-    labels = label(image, background=0)
-    number = np.max(labels) + 1
+    labels, number = label(out, background=0, return_num=True)
```
Comment (grlee77) [inline skimage/segmentation/_clear_border.py:107]: nice. thanks for refactoring this section

File: skimage/segmentation/_clear_border.py [hunk 2]
```diff
@@ -58,49 +67,54 @@ def clear_border(labels, buffer_size=0, bgval=0, in_place=False, mask=None):
...
     borders_indices = np.unique(labels[borders])
     indices = np.arange(number + 1)
     # mask all label indices that are connected to borders
     label_mask = np.in1d(indices, borders_indices)
     # create mask for pixels to clear
-    mask = label_mask[labels.ravel()].reshape(labels.shape)
-
-    if not in_place:
-        image = image.copy()
+    mask = label_mask[labels.reshape(-1)].reshape(labels.shape)
```
Comment (grlee77) [inline skimage/segmentation/_clear_border.py:115]: Is there a benefit to this `reshape(-1)` form vs. ravel? I think neither makes a copy, right? (IIRC `flatten` copies, but `ravel` does not)

**Review by grlee77 — APPROVED** [review]

File: skimage/segmentation/_clear_border.py (hunk 2 above)
Comment (rfezzani) [reply skimage/segmentation/_clear_border.py:115]: That's just what [`ravel`](https://numpy.org/doc/stable/reference/generated/numpy.ravel.html) doc says in the note :slightly_smiling_face:. It is really not important, but well, it became one of my habits... I can revert it if you wish.

Comment (grlee77) [reply skimage/segmentation/_clear_border.py:115]: I don't have a strong preference either way.

**Comment by emmanuelle** [comment]
thanks @rfezzani!

**Pull request merged**